tfjs-backend-webgpu/src/dilation_webgpu.ts

```diff
@@ -0,0 +1,78 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+
+import {backend_util} from '@tensorflow/tfjs-core';
+
+import {getMainHeaderString as main, WebGPUProgram} from './webgpu_program';
+import {computeDispatch} from './webgpu_util';
+
+export class Dilation2DProgram implements WebGPUProgram {
+  outputShape: number[];
+  shaderKey: string;
+  dispatchLayout: {x: number[], y: number[], z: number[]};
+  dispatch: [number, number, number];
+  variableNames = ['x', 'w'];
+  uniforms =
+      'filterDims: vec2<i32>, pad: vec2<i32>, stride: vec2<i32>, dilation: vec2<i32>';
+  workgroupSize: [number, number, number] = [8, 16, 1];
+
+  constructor(convInfo: backend_util.Conv2DInfo) {
+    this.outputShape = convInfo.outShape;
+    this.dispatchLayout = {x: [3], y: [1, 2], z: [0]};
```

Comment:
Please use the flat dispatch layout if your algorithm is irrelevant with compute work group layout.